### PR TITLE
Increase privacy of masked config

### LINKED ISSUE #2929: make support string more secure
Opening this just as a reference.

When a support string is generated the masked entries still show certain info:

- length of a password, user, database name in the DB URL
- length of smtp username
- domain length and TLD length, if subdomain is used or a sub directory
- length and format of smtp_from

This change masks the fields with `***` as it is done for passwords.

/ref #2878

### PR DESCRIPTION
**Pull request by BlackDex**
This changes the masking function to hide a bit more information from the generated support string. It will still keep showing the `://` for example, and `,`, but other characters will be hidden.

Also did some small changes on some key's which all showed up as `Internal` on the Settings page.

Fixes #2929

### CODE REVIEW

File: src/config.rs [hunk 1]
```diff
@@ -232,13 +232,17 @@ macro_rules! make_config {
...
+                    let mut collon_match = false;
+                    value.chars().map(|c| {
+                        n += 1;
                         match c {
-                            c if c.is_alphanumeric() => '*',
-                            '_' => '*',
-                            '-' => '*',
-                            _ => c
+                            ':' if n <= 11  => { collon_match = true ; c },
+                            '/' if n <= 13 && collon_match  => c,
```
Comment (dani-garcia) [inline src/config.rs:241]: collon -> colon :)

Comment (BlackDex) [reply src/config.rs:241]: I have to digest this a bit 🤣

**Comment by BlackDex** [comment]
@dani-garcia Done :)

File: src/config.rs [hunk 2]
```diff
@@ -232,13 +232,17 @@ macro_rules! make_config {
...
+                    let mut colon_match = false;
+                    value.chars().map(|c| {
+                        n += 1;
                         match c {
-                            c if c.is_alphanumeric() => '*',
-                            '_' => '*',
-                            '-' => '*',
-                            _ => c
+                            ':' if n <= 11  => { colon_match = true ; c },
+                            '/' if n <= 13 && colon_match  => c,
```
Comment (tessus) [inline src/config.rs:241]:
```suggestion
                            '/' if n <= 13 && colon_match => c,
```

removed extra space, unless this was intentional

Comment (BlackDex) [reply src/config.rs]: It looks like `cargo fmt` doesn't work inside macro's, not even for _simple_ code.
But i agree, that should be different. I have run `fmt` on that part manually, and replaced it with the right formatted code.

**Pull request merged**